File: backend/src/main/java/com/festago/zfestival/dto/FestivalDetailResponse.java
package com.festago.zfestival.dto;

import com.festago.stage.domain.Stage;
import com.festago.zfestival.domain.Festival;
import java.time.LocalDate;
import java.util.List;

public record FestivalDetailResponse(
    Long id,
    String name,
    LocalDate startDate,
    LocalDate endDate,
    String thumbnail,
    List<FestivalDetailStageResponse> stages) {

    public static FestivalDetailResponse of(Festival festival, List<Stage> stages) {
        List<FestivalDetailStageResponse> stageResponses = stages.stream()
            .map(FestivalDetailStageResponse::from)
            .toList();
        return new FestivalDetailResponse(
            festival.getId(),
            festival.getName(),
            festival.getStartDate(),
            festival.getEndDate(),
            festival.getThumbnail(),
            stageResponses
        );
    }
}

File: android/festago/data/src/main/java/com/festago/festago/data/dto/festival/FestivalDetailResponse.kt
package com.festago.festago.data.dto.festival

import com.festago.festago.data.dto.school.SchoolResponse
import com.festago.festago.data.dto.school.SocialMediaResponse
import com.festago.festago.data.dto.stage.StageResponse
import com.festago.festago.domain.model.festival.FestivalDetail
import kotlinx.serialization.Serializable
import java.time.LocalDate

@Serializable
data class FestivalDetailResponse(
    val id: Long,
    val name: String,
    val startDate: String,
    val endDate: String,
    val posterImageUrl: String,
    val school: SchoolResponse,
    val socialMedias: List<SocialMediaResponse>,
    val stages: List<StageResponse>,
) {
    fun toDomain() = FestivalDetail(
        id,
        name,
        LocalDate.parse(startDate),
        LocalDate.parse(endDate),
        posterImageUrl,
        school.toDomain(),
        socialMedias.map { it.toDomain() },
        stages.map { it.toDomain() },
    )
}
